pkg/plugins/backendplugin/grpcplugin/client_proto.go

```diff
@@ -66,57 +80,80 @@ func (r *protoClient) Logger() log.Logger {
 }
 
 func (r *protoClient) Start(ctx context.Context) error {
-	return r.plugin.Start(ctx)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	err := r.plugin.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	r.running = true
+	return nil
 }
 
 func (r *protoClient) Stop(ctx context.Context) error {
-	return r.plugin.Stop(ctx)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	err := r.plugin.Stop(ctx)
+	if err != nil {
+		return err
+	}
+
+	r.running = false
+	return nil
+}
+
+func (r *protoClient) readyForQuery() bool {
+	r.mu.RLock()
+	ready := r.running && r.plugin.pluginClient != nil
+	r.mu.RUnlock()
+	return ready
 }
 
 func (r *protoClient) QueryData(ctx context.Context, in *pluginv2.QueryDataRequest, opts ...grpc.CallOption) (*pluginv2.QueryDataResponse, error) {
-	if r.plugin.pluginClient == nil {
+	if r.readyForQuery() {
```

Comment:
Shouldn't it be not here? Maybe a test would be good to verify?

```suggestion
	if !r.readyForQuery() {
```

Maybe a more cleaner way would be to retrieve the client via method instead of checking `readyForQuery` to return the client or an error?

```go
c, err := r.getClient()
if err != nil {
  return nil, err
}

return c.DataClient.QueryData(ctx, in, opts...)
```